src/BenchmarkDotNet/Columns/BaselineAllocationRatioColumn.cs

```diff
@@ -0,0 +1,88 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using BenchmarkDotNet.Diagnosers;
+using BenchmarkDotNet.Mathematics;
+using BenchmarkDotNet.Reports;
+using BenchmarkDotNet.Running;
+using JetBrains.Annotations;
+
+namespace BenchmarkDotNet.Columns
+{
+    public class BaselineAllocationRatioColumn : BaselineCustomColumn
+    {
+        public override string Id => nameof(BaselineAllocationRatioColumn);
+        public override string ColumnName => "Alloc Ratio";
+
+        public override string GetValue(Summary summary, BenchmarkCase benchmarkCase, Statistics baseline, IReadOnlyDictionary<string, Metric> baselineMetrics,
+            Statistics current, IReadOnlyDictionary<string, Metric> currentMetrics, bool isBaseline)
+        {
+            double? ratio = GetAllocationRatio(currentMetrics, baselineMetrics);
+            double? invertedRatio = GetAllocationRatio(baselineMetrics, currentMetrics);
+
+            if (ratio == null)
+                return "NA";
+
+            var cultureInfo = summary.GetCultureInfo();
+            var ratioStyle = summary?.Style?.RatioStyle ?? RatioStyle.Value;
+
+            bool advancedPrecision = IsNonBaselinesPrecise(summary, baselineMetrics, benchmarkCase);
+            switch (ratioStyle)
+            {
+                case RatioStyle.Value:
+                    return ratio.Value.ToString(advancedPrecision ? "N3" : "N2", cultureInfo);
+                case RatioStyle.Percentage:
+                    return isBaseline
+                        ? ""
+                        : ratio.Value >= 1.0
+                            ? "+" + ((ratio.Value - 1.0) * 100).ToString(advancedPrecision ? "N1" : "N0", cultureInfo) + "%"
+                            : "-" + ((1.0 - ratio.Value) * 100).ToString(advancedPrecision ? "N1" : "N0", cultureInfo) + "%";
+                case RatioStyle.Trend:
+                    return isBaseline
+                        ? ""
+                        : ratio.Value >= 1.0
+                            ? ratio.Value.ToString(advancedPrecision ? "N3" : "N2", cultureInfo) + "x more"
+                            : invertedRatio == null
+                                ? "NA"
+                                : invertedRatio.Value.ToString(advancedPrecision ? "N3" : "N2", cultureInfo) + "x less";
+                default:
+                    throw new ArgumentOutOfRangeException(nameof(summary), ratioStyle, "RatioStyle is not supported");
```

Comment:
could you please refactor this logic to avoid code duplication with `BaselineRatioColumn`?

Reply:
@adamsitnik, I think you missed with lines a bit. You highlighted the default case. You meant about `IsNonBaselinesPrecise` or the whole switch or the switch math `(value - 1.0) * 100` reuse?

Reply:
@YegorStepanov the whole switch

Reply:
@adamsitnik It's hard.
`Mean` and `Allocation` differs by:
- 'Ratio' type is `Statistics` and `double?`
- one prints `"x slower/faster"`, another prints `x more/less`
- one prints '"baseline"', another prints '""'

I've tried do it thought a class with `FromValue`/`FromPercentage`/`FromTrend` methods (each method has switch for `Mean`/`StdDev`/`Allocation` column). But I can't pass `InvertedRatio', because types are different. (we should check for null, and call `.Value` for both cases.

Also, the code is hard to read.

After I've decided implement `Strategy` pattern. To convert switches to structures I've added `IRatioMetricStrategy`, `RatioMeanStrategy`, `RatioStdDevStrategy`, `RatioAllocationStrategy` (120 lines of code). But this code converts switch to methods only! To decrease duplication, we should add another strategy `IRatioStrategy` with `IMeanStrategy` and `IStdStrategy`. I think I'm wasting time :)

Maybe I don't see a simple solution.

Reply:
@YegorStepanov thanks for taking a look into this. You have convinced me that it's not worth the additional complexity ;)